File: CS/CustomGridResourceText/CustomGridLocalizer.cs
using System;
using DevExpress.Xpf.Grid;

namespace CustomGridResourceText {
    public class CustomGridLocalizer : GridControlResXLocalizer {
        public override string GetLocalizedString(GridControlStringId id) {
            if(id == GridControlStringId.GridNewRowText)
                return "Click here to add a new task";
            
            if(id == GridControlStringId.GridGroupPanelText)
                return "Here goes custom text for the group panel";

            return base.GetLocalizedString(id);
        }
    }
}

File: VB/CustomGridResourceText/CustomGridLocalizer.vb
Imports DevExpress.Xpf.Grid

Namespace CustomGridResourceText

    Public Class CustomGridLocalizer
        Inherits GridControlResXLocalizer

        Public Overrides Function GetLocalizedString(ByVal id As GridControlStringId) As String
            If id = GridControlStringId.GridNewRowText Then Return "Click here to add a new task"
            If id = GridControlStringId.GridGroupPanelText Then Return "Here goes custom text for the group panel"
            Return MyBase.GetLocalizedString(id)
        End Function
    End Class
End Namespace
